src/service/auth/signup.ts

```
@@ -1,8 +1,9 @@
 import { prisma, Role } from '../../config/prisma';
 import { Request, Response } from 'express';
 import bcrypt from 'bcrypt';
+import { SignUpRequest } from '../../types/auth';
 
-export const signUp = async (req: Request, res: Response) => {
+export const signUp = async (req: Request<{}, {}, SignUpRequest>, res: Response) => {
```

Biome (1.9.4)

[error] Don't use '{}' as a type.

Prefer explicitly define the object shape. '{}' means "any non-nullable value".

(lint/complexity/noBannedTypes)